### PR TITLE
Tweaks for Tor nightly package testing

### LINKED ISSUE #2536: Allow over-ride of Tor repository URL for testing
# Feature request

## Description

Allow over-ride of the default Tor repository used to download `tor`. This will allow us to expand testing and to move into other distributions down the road.

## User Stories

As a developer, I'd like the ability to modify where `tor` is being installed from without a bunch of manual manipulation of the ansible playbooks.

### PR DESCRIPTION
**Pull request by msheiny**
## Status

Ready for review

## Description of Changes

Contains code necessary to implement daily tor package tests against the tor nightly trusty repo. 

Subsequently also cleans up the ec2 logic and testinfra story surrounding the molecule `aws` scenario. Particularly, a developer no longer needs to specify a `vpc_id` but can utilize VPC names that default to FPF infra but can easily be over-ridden by developers testing against a personal EC2 environment.

Fixes #2536 

Changes proposed in this pull request:

## Testing

How should the reviewer test this PR?

Reviewer can test firing up an AWS environment locally if they'd like (not a requirement for the PR as far as I see it). You should check out the relevant jenkins jobs at https://ci.freedom.press/job/SD/job/tor-nightly-test/ though.

## Deployment

Nopeee - only affects CI

## Checklist

N/A

### CODE REVIEW

File: devops/jenkins/TorNightlyPipeline [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        CI_AWS_PUBLIC_IP = "no"
+        CI_SLACK_CHANNEL = "#securedrop"
+        ANSIBLE_INVENTORY = "molecule/aws/.molecule/ansible_inventory.yml"
+        FPF_GRSEC = "false" }
+
+    stages{
+        stage('Install with Tor nightly repo') {
+            steps {
+                deleteDir()
+                git branch: 'ExtendTorURL', url: 'https://github.com/freedomofpress/securedrop.git'
```
Comment (msheiny) [inline devops/jenkins/TorNightlyPipeline]: Note that this branch needs to be changed to `develop` before this PR is merged

**Comment by conorsch** [comment]
@msheiny Let's update the branch name to `develop` and get this in.

**Comment by msheiny** [comment]
@conorsch fixed the dangling CI issue here :)

**Comment by conorsch** [comment]
@msheiny The git commit author info on https://github.com/freedomofpress/securedrop/pull/2537/commits/5e280e85a9d5fab75ade574207e60b95cab071c2 looks screwed up, can you amend here? Will circle back on this after 0.5 is out the door.

**Comment by conorsch** [comment]
@msheiny Latest commit is still showing as "Unverified":

scary-unverified-commit

Try a `git reset --mixed <sha>` for that commit, then you can re-commit it with clean git config. Also, looks like the bad authorship info was introduced via these lines:

```
grep config devops/scripts/rebase-develop.sh
  # Git will yell at you when trying to rebase without author/email configured
  git config --global user.email "ci@freedom.press"
  git config --global user.name "CI User"
```

Probably worth experimenting with removing `--global` from those configs and seeing if it still works well enough for CI.

**Comment by conorsch** [comment]
Thanks, @msheiny. Will give this a review tomorrow and try to get it in this week.

**Comment by msheiny** [comment]
rebased on develop to address merge conflicts

**Pull request merged**